boa/src/syntax/parser/statement/switch/mod.rs

```diff
@@ -54,7 +54,10 @@ impl TokenParser for SwitchStatement {
         let (cases, default) =
             CaseBlock::new(self.allow_yield, self.allow_await, self.allow_return).parse(cursor)?;
 
-        Ok(Node::switch::<_, _, _, Node>(condition, cases, default))
+        match default {
+            Some(s) => Ok(Switch::new(Box::new(condition), cases, Some(Box::new(s))).into()),
+            None => Ok(Switch::new(Box::new(condition), cases, None).into()),
+        }
```

Comment:
A couple of things here. It should return a `Switch` instead of a `Node`. And we don't need a `match here:
```suggestion
        Ok(Switch::new(condition, cases, default))
```

This example above requires some generics in the `new()` function of `Switch`, as explained above.

Reply:
Fixed as part of commit ea45b261 (to be merged), found that there was already a Case struct created for a similar purpose as part of SwitchStatement parsing so removed that struct and replaced it with the new Case struct.

Reply:
Fixed